content/docs/administration/troubleshoot/databasecomponentunreadyreplicas.md

```diff
@@ -0,0 +1,200 @@
+
+{{< /details >}}
+
+{{< details "Symptom 4: Database process fails to join the database" >}}
+
+Once started, a database process communicates with the rest of the database and executes an entry protocol.
+
+Possible causes for unsuccessful startup during this phase are:
+
+- Network issues prevent communication between NuoDB database processes
+- No suitable entry node is available
+- The database process binary version is too old
+
+{{< /details >}}
+
+{{< details "Symptom 5: An SM in `TRACKED` state for a long time" >}}
+
+The database state might be `AWAITING_ARCHIVE_HISTORIES_MSG` indicating that the database leader assignment is in progress.
+NuoDB Admin must collect archive history information from all provisioned archives on database cold start.
+This requires all SM processes to start and connect to the NuoDB Admin within the configured timeout period.
+
+Possible causes for unsuccessful leader assignment:
+
+- Not all SMs have been scheduled by Kubernes or not all SM processes have started
+- Some of the SM pods are in `CrashLoopBackOff` state with long back-off
+- There is a _ghost_ archive metadata provisioned in the domain which is not served by an actual SM
+
+{{< /details >}}
+
+{{< details "Symptom 6: An TE in `TRACKED` state for a long time" >}}
+
+A TE process joins the database via an entry node which is normally the first SM that goes to `RUNNING` state.
+NuoDB Admin performs synchronization tasks so that TEs are started after the entry node is available.
+
+Possible causes for missing entry node:
+
+- Database leader assignment is not performed after cold start. See _Symptom 5_
+- The `UNPARTITIONED` storage group is not in `RUNNING` state
+
+{{< /details >}}
+
+{{< details "Symptom 7: SM in `CONFIGURED:RECOVERING_JOURNAL` state for a long time" >}}
+
+Upon startup, SM processes perform a journal recovery procedure by applying any transaction messages to the atoms.
+This involves extensive disk IO and may continue for a while depending on the backlog of messages and the number of atoms to which they are applied.
+The SM process reports the progress of the journal recovery which is displayed in `nuocmd show domain` output.
+
+Possible causes for slow journal recovery:
+
+- High latency of the archive disk caused by reaching the IOPS limit
+
+{{< /details >}}
+
+### Example
+
+Get the database name and its namespace from the alert's labels.
+Inspect the database state in the Kubernetes cluster.
+
+```sh
+kubectl get database acme-messaging-demo -n nuodb-cp-system
+```
+
+Notice that the `READY` status condition is `False` which means that the database is in a degraded state.
+
+```text
+NAME                  TIER       VERSION   READY   SYNCED   DISABLED   AGE
+acme-messaging-demo   n0.small   6.0.2     False   True     False      46h
+```
+
+Inspect the database components state.
+
+```sh
+kubectl get database acme-messaging-demo -o jsonpath='{.status.components}' | jq
```

Comment:
`jq` is only used for pretty-printing?

Reply:
Right. I couldn't find similar option for `kubectl`.